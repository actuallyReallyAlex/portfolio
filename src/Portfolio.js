import React, { Component } from 'react'
import PropTypes from 'prop-types'
// import NavMenu from './components/NavMenu'
// import { Intro } from './components/Intro'
import { App } from 'grommet'
import NavMenu from './components/NavMenu'
import Intro from './sections/Intro'
import About from './sections/About'

class Portfolio extends Component {
  render() {
    return (
      <App>
        <NavMenu />
        <Intro />
        <About />
      </App>
    )
  }
}

Portfolio.propTypes = {
  themeColor: PropTypes.string
}

export default Portfolio
