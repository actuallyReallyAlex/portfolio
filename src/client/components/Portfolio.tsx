import * as React from "react";
import { Link } from "react-router-dom";

import { PortfolioItemDocument } from "../types";

export interface PortfolioProps {
  portfolioItems: PortfolioItemDocument[];
  setPortfolioItems: (portfolioItems: PortfolioItemDocument[]) => void;
}

const Portfolio: React.FunctionComponent<PortfolioProps> = (
  props: PortfolioProps
) => {
  const { portfolioItems, setPortfolioItems } = props;
  React.useEffect(() => {
    const getPortfolioItems = async (): Promise<
      PortfolioItemDocument[] | { error: any }
    > => {
      try {
        const response = await fetch("/portfolioItems", {
          headers: {
            "Content-Type": "application/json",
          },
        });
        const data = await response.json();

        setPortfolioItems(data);
      } catch (error) {
        console.error(error);
        return { error };
      }
    };
    getPortfolioItems();
  }, []);

  return (
    <div>
      <h2>Portfolio</h2>
      <ul>
        {portfolioItems.map((item) => (
          <li key={item._id}>
            <div>
              <h3>{item.title}</h3>
              <h4>{item.tagline}</h4>
              <Link to={`/portfolio/${item._id}`}>View Item</Link>
            </div>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default Portfolio;
