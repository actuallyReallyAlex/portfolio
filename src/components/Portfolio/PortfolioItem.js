import React from 'react'
import PropTypes from 'prop-types'
import { Anchor, Box, Heading, Text } from 'grommet'
import { Code, LinkNext } from 'grommet-icons'
import Tilt from 'react-tilt'

const PortfolioItem = ({
  category,
  description,
  image,
  languages,
  link,
  size,
  source,
  title
}) => {
  return (
    <Tilt
      className="Tilt"
      options={{ max: 15, perspective: 2000, scale: 1.05 }}
    >
      <Box
        background="brand"
        className="Tilt-inner"
        width={size === 'small' ? '300' : '384px'}
      >
        <Box
          background={`url(${image})`}
          height="189px"
          width={size === 'small' ? '300' : '384px'}
        />
        <Box pad="medium">
          <Text size="medium">{category}</Text>
          <Text size="xlarge" weight="bold">
            {title}
          </Text>
          <Box margin={{ vertical: 'medium' }}>
            {languages.map(({ color, title }, i) => (
              <Box
                align="center"
                direction="row"
                gap="small"
                key={i}
                justify="start"
              >
                <Box background={color} height="10px" width="10px" />
                <Heading level="4" margin="none">
                  {title}
                </Heading>
              </Box>
            ))}
          </Box>
          <Box fill="horizontal">
            <Text>{description}</Text>
          </Box>
          <Box margin={{ top: 'medium' }}>
            {source && (
              <Box direction="row" gap="small">
                <Anchor
                  color="white"
                  href={source.href}
                  icon={<Code />}
                  label={source.title}
                  rel="noreferrer"
                  target="_blank"
                />
              </Box>
            )}
            {link && (
              <Box direction="row" gap="small">
                <Anchor
                  color="white"
                  href={link.href}
                  icon={<LinkNext />}
                  label={link.title}
                  rel="noreferrer"
                  target="_blank"
                />
              </Box>
            )}
          </Box>
        </Box>
      </Box>
    </Tilt>
  )
}

PortfolioItem.propTypes = {
  category: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  languages: PropTypes.array.isRequired,
  link: PropTypes.object,
  size: PropTypes.string.isRequired,
  source: PropTypes.object,
  title: PropTypes.string.isRequired
}

export default PortfolioItem
