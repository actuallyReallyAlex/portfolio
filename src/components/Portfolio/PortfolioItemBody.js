import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { Box, Text } from 'grommet'

const PortfolioItemBody = ({ description, languages }) => {
  return (
    <Fragment>
      <Box margin={{ vertical: 'medium' }}>
        {languages.map(({ color, title }, i) => (
          <Box
            align="center"
            direction="row"
            gap="small"
            key={i}
            justify="start"
          >
            <Box background={color} height="10px" width="10px" />
            <Text margin="none" weight="bold">
              {title}
            </Text>
          </Box>
        ))}
      </Box>
      <Box fill="horizontal">
        <Text>{description}</Text>
      </Box>{' '}
    </Fragment>
  )
}

PortfolioItemBody.propTypes = {
  description: PropTypes.string.isRequired,
  languages: PropTypes.array.isRequired
}

export default PortfolioItemBody
