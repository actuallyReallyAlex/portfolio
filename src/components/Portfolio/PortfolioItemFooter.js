import React from 'react'
import PropTypes from 'prop-types'
import { Anchor, Box } from 'grommet'
import { Code, LinkNext } from 'grommet-icons'

const PortfolioItemFooter = ({ link, source }) => {
  return (
    <Box margin={{ top: 'medium' }}>
      {source && (
        <Box direction="row" gap="small">
          <Anchor
            color="white"
            href={source.href}
            icon={<Code />}
            label={source.title}
            rel="noreferrer"
            target="_blank"
          />
        </Box>
      )}
      {link && (
        <Box direction="row" gap="small">
          <Anchor
            color="white"
            href={link.href}
            icon={<LinkNext />}
            label={link.title}
            rel="noreferrer"
            target="_blank"
          />
        </Box>
      )}
    </Box>
  )
}

PortfolioItemFooter.propTypes = {
  link: PropTypes.object,
  source: PropTypes.object
}

export default PortfolioItemFooter
