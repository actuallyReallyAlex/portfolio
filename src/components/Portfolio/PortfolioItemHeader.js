import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import { Text } from 'grommet'

const PortfolioItemHeader = ({ category, title }) => {
  return (
    <Fragment>
      <Text size="medium">{category}</Text>
      <Text size="xlarge" weight="bold">
        {title}
      </Text>
    </Fragment>
  )
}

PortfolioItemHeader.propTypes = {
  category: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}

export default PortfolioItemHeader
