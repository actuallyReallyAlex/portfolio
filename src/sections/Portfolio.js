import React from 'react'
import { Section, Tiles, Tile } from 'grommet'
import { BrowserView, MobileView } from 'react-device-detect'
import SectionHeader from '../components/SectionHeader'
import Tilty from '../components/Tilty'

// Works
import Reposier from '../components/works/Reposier'
import Splague from '../components/works/Splague'
import PayrollAdmin from '../components/works/PayrollAdmin'
import PayrollForm from '../components/works/PayrollForm'
import StaffingDashboard from '../components/works/StaffingDashboard'
import Contractor from '../components/works/Contractor'
import StaffWebsite from '../components/works/StaffWebsite'
import FacilitatorWebsite from '../components/works/FacilitatorWebsite'
import Remembera from '../components/works/Remembera'
import Pickitt from '../components/works/Pickitt'
import PortfolioWork from '../components/works/PortfolioWork'
import PortfolioDocs from '../components/works/PortfolioDocs'

const Portfolio = () => {
  return (
    <Section id="portfolio">
      <SectionHeader
        title="Portfolio"
        subtitle="Check out some of my projects."
      />
      <BrowserView>
        <Tiles flush={false} fill>
          <Tilty portfolioItem item={<Reposier />} />
          <Tilty portfolioItem item={<Splague />} />
          <Tilty portfolioItem item={<PayrollAdmin />} />
          <Tilty portfolioItem item={<PayrollForm />} />
          <Tilty portfolioItem item={<Remembera />} />
          <Tilty portfolioItem item={<Pickitt />} />
          <Tilty portfolioItem item={<StaffingDashboard />} />
          <Tilty portfolioItem item={<Contractor />} />
          <Tilty portfolioItem item={<StaffWebsite />} />
          <Tilty portfolioItem item={<FacilitatorWebsite />} />
          <Tilty portfolioItem item={<PortfolioWork />} />
          <Tilty portfolioItem item={<PortfolioDocs />} />
        </Tiles>
      </BrowserView>
      <MobileView>
        <Tile>
          <Reposier />
        </Tile>
        <Tile>
          <Splague />
        </Tile>
        <Tile>
          <PayrollAdmin />
        </Tile>
        <Tile>
          <PayrollForm />
        </Tile>
        <Tile>
          <Remembera />
        </Tile>
        <Tile>
          <Pickitt />
        </Tile>
        <Tile>
          <StaffingDashboard />
        </Tile>
        <Tile>
          <Contractor />
        </Tile>
        <Tile>
          <StaffWebsite />
        </Tile>
        <Tile>
          <FacilitatorWebsite />
        </Tile>
        <Tile>
          <PortfolioWork />
        </Tile>
        <Tile>
          <PortfolioDocs />
        </Tile>
      </MobileView>
    </Section>
  )
}

export default Portfolio
