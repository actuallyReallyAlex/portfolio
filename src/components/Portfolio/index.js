import React from 'react'
import { Box } from 'grommet'
import SectionHeading from '../SectionHeading'
import { portfolio } from '../../config.json'
import PortfolioItem from './PortfolioItem'

const Portfolio = () => {
  return (
    <Box
      align="center"
      fill="horizontal"
      gap="medium"
      id="Portfolio"
      justify="center"
      pad="large"
    >
      <SectionHeading
        heading={portfolio.heading}
        subHeading={portfolio.subHeading}
      />
      {portfolio.rows.map(({ column1, column2 }, i) => {
        return (
          <Box
            align="start"
            direction="row"
            fill="horizontal"
            justify="center"
            key={i}
            gap="xlarge"
            margin={{ vertical: 'large' }}
          >
            <PortfolioItem
              category={column1.category}
              description={column1.description}
              image={column1.image}
              languages={column1.languages}
              link={column1.link}
              source={column1.source}
              title={column1.title}
            />
            <PortfolioItem
              category={column2.category}
              description={column2.description}
              image={column2.image}
              languages={column2.languages}
              link={column2.link}
              source={column2.source}
              title={column2.title}
            />
          </Box>
        )
      })}
    </Box>
  )
}

export default Portfolio
