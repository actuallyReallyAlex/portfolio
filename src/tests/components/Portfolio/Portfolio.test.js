import React from 'react'
import Portfolio from '../../../components/Portfolio'
import { shallow } from 'enzyme'

const defaultProps = {
  navRef: jest.fn(),
  size: 'medium'
}

describe('<Portfolio />', () => {
  test('Should render the <Portfolio />.', () => {
    const component = shallow(<Portfolio {...defaultProps} />)
    expect(component).toMatchSnapshot()
  })
})
