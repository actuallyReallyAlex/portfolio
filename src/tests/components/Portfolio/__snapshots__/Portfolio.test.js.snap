// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Portfolio /> Should render the <Portfolio />. 1`] = `ShallowWrapper {}`;
